jobs/jenkins-master/templates/config/configure-devsystems-creds.groovy.erb

```diff
@@ -0,0 +1,30 @@
+import jenkins.model.Jenkins
+import com.cloudbees.plugins.credentials.CredentialsScope
+import com.cloudbees.plugins.credentials.domains.Domain
+import com.cloudbees.jenkins.plugins.sshcredentials.impl.BasicSSHUserPrivateKey
+import com.cloudbees.jenkins.plugins.sshcredentials.impl.BasicSSHUserPrivateKey.DirectEntryPrivateKeySource
+
+final String SYSTEM_CREDENTIALS_PROVIDER_CLASSPATH = 'com.cloudbees.plugins.credentials.SystemCredentialsProvider'
+
+final String GHE_DEVSYSTEMS_SSH_KEY = '''<%= p('jenkins.github.devsystems.ssh') %>'''
+
+globalDomain = Domain.global()
+
+if (GHE_DEVSYSTEMS_SSH_KEY.allWhitespace) {
+  println 'jenkins.github.devsystems.ssh has no content'
+
+  System.exit(1)
```

Comment:
Is this likely to fail the deployment if an ssh key is not provided? If so, is that the expected behaviour should you wish to deploy a standalone jenkins instance?

Reply:
Dunno if you saw the message on Slack, but please don't merge these tonight. They were prepped for discussion tomorrow.

Anyway ... it's a good point. I was thinking you'd use ${JENKINS_USE_GITHUB_AUTH} (checked in pre-start) to prevent the script even being invoked if you didn't want to use GitHub, but it's a bit indirect. Would be good to know your thoughts tomorrow about what we should do in such error scenarios.